### PR TITLE
Rewrite Chapter 14: Entities and Resolution

### LINKED ISSUE #120: 14: Consider rewriting whole chapter
In chapter **14. Names and Resolution**

*Issue exported from our legacy tracking system.*

### PR DESCRIPTION
**Pull request by kirtchev-adacore**
Fixes #120

### CODE REVIEW

**Comment by pietroalbini** [comment]
You need to add the file to the index.

File: src/entities-and-resolution.rst [hunk 1]
```diff
@@ -0,0 +1,1704 @@
...
+
+* :dp:`fls_lxa7uhmdoy9d`
+  If the leftmost :t:`path segment` starts with a :t:`qualified type`, then
+
+  * :dp:`fls_xujlscsir05f`
+    If the :t:`qualified type` is subject to a :t:`qualifying trait`, then the
+    :t:`resolution context` of the leftmost :t:`path segment` consists of the
+    :t:`entities <entity>` of all :t:`[implementation]s` of the
+    :t:`qualified type` that implement the :t:`qualifying trait`.
+    **What does the path segment resolve to?**
```
Comment (Veykril) [inline src/entities-and-resolution.rst:1444]: to a type

File: src/entities-and-resolution.rst [hunk 2]
```diff
@@ -0,0 +1,1704 @@
...
+
+  * :dp:`fls_wypnvfklnmc1`
+    Otherwise the :t:`resolution context` of the leftmost :t:`path segment`
+    consists of the :t:`entity` of the :t:`qualified type`, the
+    :t:`entities <entity>` of all its :t:`[inherent implementation]s`, and the
+    :t:`entities <entity>` of all its :t:`[trait implementation]s` of
+    :t:`[trait]s` that are :t:`in scope`.
+    **What does the path segment resolve to?**
```
Comment (Veykril) [inline src/entities-and-resolution.rst:1452]: to a type

File: src/entities-and-resolution.rst [hunk 3]
```diff
@@ -0,0 +1,1704 @@
...
+.. _fls_40xoego2thsp:
+
+Resolution
+----------
+
+.. rubric:: Legality Rules
+
+:dp:`fls_ho4kem1slcxg`
+:t:`Resolution` is the process of finding a unique interpretation for a
+:t:`construct` in a program.
```
[the remark this answers is not in the document]
Comment (kirtchev-adacore) [reply src/entities-and-resolution.rst]: Good idea, done!

File: src/entities-and-resolution.rst (hunk 1 above)
Comment (kirtchev-adacore) [reply src/entities-and-resolution.rst]: Fixed.

File: src/entities-and-resolution.rst (hunk 2 above)
Comment (kirtchev-adacore) [reply src/entities-and-resolution.rst]: Fixed.

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
bors merge

**Pull request merged**